ariadne/contrib/federation/schema.py

```diff
@@ -76,25 +76,27 @@ def make_federated_schema(
         r"(?<=@link).*?url:.*?\"(.*?)\".?[^)]+?", sdl, re.MULTILINE | re.DOTALL
     )  # use regex to parse if it's fed 1 or fed 2; adds dedicated typedefs per spec
     dirname = os.path.dirname(__file__)
-    if link and link.group(1) == "https://specs.apollo.dev/federation/v2.0":
-        definitions = load_schema_from_path(
-            os.path.join(dirname, "./definitions/fed2_0.graphql")
-        )
-    elif link and link.group(1) == "https://specs.apollo.dev/federation/v2.1":
-        definitions = load_schema_from_path(
-            os.path.join(dirname, "./definitions/fed2_1.graphql")
-        )
-    elif link and link.group(1) == "https://specs.apollo.dev/federation/v2.2":
-        definitions = load_schema_from_path(
-            os.path.join(dirname, "./definitions/fed2_2.graphql")
-        )
-    elif link and link.group(1) == "https://specs.apollo.dev/federation/v2.3":
-        definitions = load_schema_from_path(
-            os.path.join(dirname, "./definitions/fed2_3.graphql")
+    definitions_folder = os.path.join(dirname, "definitions")
+    definition_files = os.listdir(definitions_folder)
+    definition_files.sort()
+    last_version_file = definition_files[-1]
+
+    if link:
+        # extract version
+        fed_version = link.group(1).split("/")[-1]
```

Comment:
Doesn't this change mean that `https://reddit.com/r/something/v2.1` will also pass as Federation v2.1?

Reply:
oh yeah! I should validate the link url and that it belongs to the fed spec. good catch! 🙏